PyInstaller/building/makespec.py

```diff
@@ -236,23 +250,17 @@ def __add_options(parser):
     g = parser.add_argument_group('What to bundle, where to search')
     g.add_argument(
         '--add-data',
-        action='append',
-        default=[],
-        type=add_data_or_binary,
-        metavar='<SRC;DEST or SRC:DEST>',
+        action=SourceDestAction,
         dest='datas',
-        help='Additional non-binary files or folders to be added to the executable. The path separator  is platform '
-        'specific, ``os.pathsep`` (which is ``;`` on Windows and ``:`` on most unix systems) is used. This option '
-        'can be used multiple times.',
+        help="Additional non-binary files or folders to be added to the executable. Specify using the path to the "
+        "source file and the folder relative to the application's root to put the file in separated by a :. "
```

Comment:
I think this formulation should be made a bit clearer for new users. Something like:

Additional data files or directories containing data files to be added to the application. The argument value should be in form of "source:dest_dir", where source is the path to file (or directory) to be collected, dest_dir is the destination directory relative to the top-level application directory, and both paths are separated by a colon (:). To put a file in the top-level application directory, use . as a dest_dir.